src/Type/Symfony/EnvelopeReturnTypeExtension.php

```diff
@@ -33,15 +35,15 @@ public function getTypeFromMethodCall(
 	): Type
 	{
 		if (count($methodCall->getArgs()) === 0) {
-			return new ArrayType(new MixedType(), new ArrayType(new MixedType(), new ObjectType('Symfony\Component\Messenger\Stamp\StampInterface')));
+			return new ArrayType(new StringType(), new ArrayType(new IntegerType(), new ObjectType('Symfony\Component\Messenger\Stamp\StampInterface')));
 		}
 
 		$argType = $scope->getType($methodCall->getArgs()[0]->value);
 		if (!$argType instanceof ConstantStringType) {
-			return new ArrayType(new MixedType(), new ObjectType('Symfony\Component\Messenger\Stamp\StampInterface'));
+			return new ArrayType(new IntegerType(), new ObjectType('Symfony\Component\Messenger\Stamp\StampInterface'));
 		}
 
-		return new ArrayType(new MixedType(), new ObjectType($argType->getValue()));
+		return new ArrayType(new IntegerType(), new ObjectType($argType->getValue()));
```

Comment:
Hmm... how can I tell PHPStan here that it's a `list<>`? :thinking:

Reply:
The test needs to run with bleedingEdge config, put it in `getAdditionalConfigFiles`.

```
'phar://' . __DIR__ . '/../../../vendor/phpstan/phpstan/phpstan.phar/conf/bleedingEdge.neon'
```

Reply:
But even then this says `array<int, ...>` instead of `list<...>` - and there do have different meaning. I don't think PHPStan should assume that every array with integer keys is a list.

Reply:
Yeah, as expected bleeding edge config didn't change anything.

Reply:
You need to call AccessoryArrayListType::intersect() on the type you want to become a list.

Reply:
I found AccessoryArrayListType but it doesn't take an argument to specify the value type so I don't think that's what I need here either.

Reply:
Just look how it's used in phpstan-src.

Reply:
Ah okay. This works when combined with bleeding edge.

```php
AccessoryArrayListType::intersectWith(new ArrayType(new IntegerType(), new ObjectType('Symfony\Component\Messenger\Stamp\StampInterface'))),
```